### PR TITLE
Move parentchain pallet under worker-pallets folder

### PR DESCRIPTION
**Pull request by Traf333**
### Context

parentchain pallet uses only by workers and within this PR we move it to dedicated folder

### CODE REVIEW

File: bitacross-worker/app-libs/sgx-runtime/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ sp-std = { default-features = false, git = "https://github.com/paritytech/substr
 sp-version = { default-features = false, git = "https://github.com/paritytech/substrate.git", branch = "polkadot-v0.9.42" }
 
 
-pallet-parentchain = { path = "../../../pallets/parentchain", default-features = false }
+pallet-parentchain = { path = "../../../tee-worker/litentry/pallets/parentchain", default-features = false }
```
Comment (kziemianek) [inline bitacross-worker/app-libs/sgx-runtime/Cargo.toml]: I think workers shouldn't depend on each other. 
What about such structure:
```
litentry-parachain/
  - pallets/
  - worker-pallets/
```
?

Comment (Traf333) [reply bitacross-worker/app-libs/sgx-runtime/Cargo.toml]: thanks, that answers my question within the task :) https://linear.app/litentry/issue/P-613/move-pallet-parentchain-to-worker-space#comment-ebe106bc

Comment (Traf333) [reply bitacross-worker/app-libs/sgx-runtime/Cargo.toml]: during to discussion, the parentchain pallet will be removed with sidechain functionality https://linear.app/litentry/issue/P-647/remove-sidechain-functionality-from-bc-worker. Within this scope I'd keep it as it is, in order to have worker compiled. What do you think @kziemianek ?

Comment (kziemianek) [reply bitacross-worker/app-libs/sgx-runtime/Cargo.toml]: I suggest merging it once `pallet-parentchain` is removed from `bitacross-worker`.
If you decide to merge it before, please update the PR description.

Comment (Traf333) [reply bitacross-worker/app-libs/sgx-runtime/Cargo.toml]: @kziemianek it seems like your proposal to move pallet to another root level folder was the easiest part. I wasn't able to remove pallet usage from bc-worker, so I've moved pallet to dedicated folder

**Review by kziemianek — APPROVED** [review]

**Pull request merged**